repertory/repertory.go

```diff
@@ -0,0 +1,81 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2020 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package repertory
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/arduino/arduino-cli/cli/feedback"
+	"github.com/gofrs/uuid"
+	"github.com/spf13/viper"
+)
+
+// Store is the Read Only config storage
+var Store = viper.New()
+
+var (
+	// Type is the repertory file type
+	Type = "yaml"
+	// Name is the repertory file Name with Type as extension
+	Name = "repertory" + "." + Type
```

Comment:
I tried the deamon and the `repertory.yaml` file is overwritten at every run (so we lose the previously generated installation id each time the deamon starts). After adding a print of the error I found that viper is unable to find the config file even if the file is there. Since we do recreate the config in case of `viper.ConfigFileNotFoundError` the file is overwritten.

Looking at the viper docs I see that `Name` should be without extension https://github.com/spf13/viper#reading-config-files and in fact by removin the extension from `Name` it works. (I glanced into viper internals and the config file handling seems a bit overengineered, but... anyway...)

Reply:
This bug is caused by an issue contained in viper 1.3.2, upgrading to 1.6.2 (the version that I had during my tests) solved your issue. 
Maybe something went wrong while rebasing the PR